Tests/DatadogTests/DatadogPrivate/URLSessionSwizzlerTests.swift

```diff
@@ -0,0 +1,410 @@
+//        }
+//        DispatchQueue.concurrentPerform(iterations: iterationCount) { iteration in
+//            do {
+//                try URLSessionSwizzler.swizzle(swizzledSessions[iteration].0, requestInterceptor: requestInterceptor, taskObserver: taskObserver)
+//                let task = swizzledSessions[iteration].0.dataTask(with: url)
+//
+//                XCTAssertNotNil(task)
+//                swizzledSessions[iteration].1.fulfill()
+//            } catch {
+//                XCTAssertNil(error)
+//            }
+//        }
+//        waitForExpectations(timeout: 0.5, handler: nil)
+//    }
+
+    // MARK: - URLSessionDataTask methods
+    func testDataTaskWithURL() {
+        let url = URL(string: "http://foo.bar")!
+        let task = setupTracedSession().dataTask(with: url)
+
+        XCTAssertEqual(task.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+
+        tearDownTracedSession()
+    }
+    func testDataTaskWithURLRequest() {
+        let url = URL(string: "http://foo.bar")!
+        let request = URLRequest(url: url)
+        let task = setupTracedSession().dataTask(with: request)
+
+        XCTAssertEqual(task.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+
+        tearDownTracedSession()
+    }
+    func testDataTaskWithURLCompletionHandler() {
+        let url = URL(string: "http://foo.bar")!
+        let task = setupTracedSession().dataTask(with: url) { _,_,_ in }
+
+        XCTAssertEqual(task.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+
+        tearDownTracedSession()
+    }
+    func testDataTaskWithURLRequestCompletionHandler() {
+        let url = URL(string: "http://foo.bar")!
+        let request = URLRequest(url: url)
+        let task = setupTracedSession().dataTask(with: request) { _,_,_ in }
+
+        XCTAssertEqual(task.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+
+        tearDownTracedSession()
+    }
+
+    func testAllDataTaskMethodsAtOnce() {
+        let url = URL(string: "http://foo.bar")!
+        let request = URLRequest(url: url)
+        let session = setupTracedSession()
+
+        let dataTaskURL = session.dataTask(with: url)
+        let dataTaskRequest = session.dataTask(with: request)
+        let dataTaskURLCompletion = session.dataTask(with: url) { _,_,_ in }
+        let dataTaskRequestCompletion = session.dataTask(with: request) { _,_,_ in }
+
+        XCTAssertEqual(dataTaskURL.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+        XCTAssertEqual(dataTaskRequest.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+        XCTAssertEqual(dataTaskURLCompletion.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+        XCTAssertEqual(dataTaskRequestCompletion.originalRequest?.allHTTPHeaderFields, injectedHeaders)
+
+        tearDownTracedSession()
+    }
+
+    func testSwizzledBackgroundSession() {
```

Comment:
Why do we need to test `.background` configuration for `URLSession` separately? It's still the same `URLSession` object, using the same API, so this test covers `URLSession` logic, not our swizzling.

Reply:
This test case was just a nice-to-have IMO, `.background` sessions don't support completion handlers and they raise an exception. I wanted to see if our swizzled methods work fine in that case as well.

(I wondered about that because I think I saw a few questions/threads somewhere regarding swizzling background session weren't working fine or something like that)

Reply:
Ah, I see 👌👍.